scripts/irods/test/session.py

```diff
@@ -124,6 +124,10 @@ def zone_name(self):
     def username(self):
         return self._environment_file_contents['irods_user_name']
 
+    @property
+    def qualified_username(self):
+        return self.username + '#' + self.zone_name
```

Comment:
This breaks two tests in test_auth.py. The tests modify environment properties which cause `self.zone_name` to throw an exception.

My proposed fix is to check if `irods_zone_name` is defined in  `self.environment_file_contents` first. If it is defined, use it. Else, return the username with a suffix of `#?`. _(I've already verified that this fixes the tests)_

Other ideas:
- return just the username
- `#<unknown>`
- `#<not-defined>`
- `#???`

Thoughts?

Reply:
not exactly sure what is breaking - but we always are operating in the context of a zone.

if one is not specified, shouldn't we always assume and use the local zonename?  of the server?

Reply:
This is specific to the test framework. The icommands/server are working as expected.

For example, one of the tests deletes the zone property from the python dict to verify iinit can handle it gracefully. When iinit is executed, the test framework attempts to log a line containing the command to execute. At this point, it falls over because it can't produce the qualified username. It can't produce the qualified username because the zone property was removed from the python dict.

Reply:
so we should log what would be used... yes?    so i think that's a vote for `just the username` or `#<not-defined>`.

since it's just the test framework, `#<not-defined>` seems good?

Reply:
I agree. I went with `#<undefined>`.

It is just an indicator that the zone was not there.